### PR TITLE
feat(config): one config file and .env file per env

### PR DESCRIPTION
**Pull request by Oupsla**
Following discussion on : https://github.com/fivethree-team/nestjs-prisma-starter/pull/420

The goal of this PR is to be able to have one config file (and one .env file) for each environnement (development, staging, ...).

Little note, I have renamed `example.env` to `.env.development` because by default we will require this file

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -21,10 +21,10 @@
     "prebuild": "rimraf dist",
     "build": "nest build",
     "format": "prettier --write \"src/**/*.ts\" \"test/**/*.ts\"",
-    "start": "nest start",
-    "start:dev": "nest start --watch",
-    "start:debug": "nest start --debug --watch",
-    "start:prod": "node dist/main",
+    "start": "NODE_ENV=${NODE_ENV:-development} nest start",
```
Comment (Oupsla) [inline package.json:24]: For info the syntax `NODE_ENV=${NODE_ENV:-development}` means that a NODE_ENV variable can be overwritten when calling `npm run` (for example `NODE_ENV=staging npm run start) but this variable will got `development` if not provided.

File: src/app.module.ts [hunk 2]
```diff
@@ -8,12 +8,19 @@ import { PostModule } from './resolvers/post/post.module';
 import { AppResolver } from './resolvers/app.resolver';
 import { DateScalar } from './common/scalars/date.scalar';
 import { ConfigModule, ConfigService } from '@nestjs/config';
-import config from './configs/config';
 import { GraphqlConfig } from './configs/config.interface';
 
+// Import config from file according to NODE_ENV
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const config = require(`./configs/${process.env.NODE_ENV}`).default;
```
Comment (Oupsla) [inline src/app.module.ts:15]: Unfortunately had to disable eslint to be able to dynamically require a config file

**Comment by Oupsla** [comment]
Hi @marcjulian , tell me if this PR is missing something 🙂

**Comment by Oupsla** [comment]
Friendly ping @marcjulian 🙂

**Comment by marcjulian** [comment]
@Oupsla I haven't had the time to look at your PR. I will come back to it 👍  thanks again for contributing to this project!

**Comment by adamrogas** [comment]
@Oupsla nest supports what you are trying to do out of the box. with file based 

.env.development
.env.local
.env.test 
{....}

@marcjulian there is actually a problem with the way the config is being loaded.

the config.ts file should not use a constant but should return the configuration from the exported function so it builds the return object when it is actually loaded (after dot env as done its magic as part of the nestjs configuration service) rather than how it is now
 
()=>{ ....config }    can load file based .env variables  

rather than 

 const config = {}      can't load file based .env variables as the constant is defined before dot-env is initialized by nestJS
 ()=> config;

**Comment by Oupsla** [comment]
> @Oupsla nest supports what you are trying to do out of the box. with file based

Yes nest loads the correct .env file based on the NODE_ENV variable (thanks to dotenv package).
But the goal was to have a config file for non-secret variables (typed, more structured, ...) and so to have one config file per environnement. 

I don't know if nest can load a config file out of the box ?

> @marcjulian there is actually a problem with the way the config is being loaded.

I didn't know about that, but you are maybe right, I'll convert it into a function that loads the correct config file 🙂

**Comment by marcjulian** [comment]
@adamrogas do you mean it should be loaded like so 

```ts
export default () => ({
  port: parseInt(process.env.PORT, 10) || 3000,
  database: {
    host: process.env.DATABASE_HOST,
    port: parseInt(process.env.DATABASE_PORT, 10) || 5432
  }
});
```

It is also described in the docs https://docs.nestjs.com/techniques/configuration#custom-configuration-files

**Pull request closed without merging**